src/Makefile

```diff
@@ -267,24 +267,40 @@ ifeq ($(MALLOC),jemalloc)
 	FINAL_LIBS := ../deps/jemalloc/lib/libjemalloc.a $(FINAL_LIBS)
 endif
 
-ifeq ($(BUILD_TLS),yes)
-	FINAL_CFLAGS+=-DUSE_OPENSSL $(OPENSSL_CFLAGS)
-	FINAL_LDFLAGS+=$(OPENSSL_LDFLAGS)
-	LIBSSL_PKGCONFIG := $(shell $(PKG_CONFIG) --exists libssl && echo $$?)
+# LIBSSL & LIBCRYPTO
+LIBSSL_LIBS=
+LIBSSL_PKGCONFIG := $(shell $(PKG_CONFIG) --exists libssl && echo $$?)
 ifeq ($(LIBSSL_PKGCONFIG),0)
 	LIBSSL_LIBS=$(shell $(PKG_CONFIG) --libs libssl)
 else
 	LIBSSL_LIBS=-lssl
 endif
-	LIBCRYPTO_PKGCONFIG := $(shell $(PKG_CONFIG) --exists libcrypto && echo $$?)
+LIBCRYPTO_LIBS=
+LIBCRYPTO_PKGCONFIG := $(shell $(PKG_CONFIG) --exists libcrypto && echo $$?)
 ifeq ($(LIBCRYPTO_PKGCONFIG),0)
 	LIBCRYPTO_LIBS=$(shell $(PKG_CONFIG) --libs libcrypto)
 else
 	LIBCRYPTO_LIBS=-lcrypto
 endif
+
+BUILD_YES:=89
+BUILD_EXT:=69
```

Comment:
I don't think i follow you.
TLS is added in two steps: one is to decide if TLS support is at all included in the build (when disabled we don't even need the headers installed on the system), and the other one decides if redis-server has it build in, or as a loadable extension.
it can be one define with 3 states (0,1,2 / undefined,1,2), or it can be two defines that are either both undefined, both defined, or just one defined.
i think the second approach is less confusing since it doesn't rely on the value of the define, and we have clear names.
the client side (redis-cli and redis-benchmark) will only be using the first define (which dictates that TLS support is at all enabled for the build).

maybe elaborate on what you meant by client, in case i'm missing something.